experimental/casemap/src/titlecase.rs

```diff
@@ -0,0 +1,401 @@
+///
+/// assert_eq!(cm.titlecase_segment_to_string("ijkdijk", &root, default_options), "Ijkdijk");
+/// assert_eq!(cm.titlecase_segment_to_string("ijkdijk", &langid!("nl"), default_options), "IJkdijk"); // Dutch IJ digraph
+/// ```
+///
+/// Head adjustment behaviors:
+///
+/// ```rust
+/// use icu_casemap::TitlecaseMapper;
+/// use icu_casemap::titlecase::{HeadAdjustment, TitlecaseOptions};
+/// use icu_locid::langid;
+///
+/// let cm_normal = TitlecaseMapper::new();
+/// let cm_legacy = TitlecaseMapper::new_legacy();
+/// let root = langid!("und");
+///
+/// let default_options = Default::default();
+/// let mut no_adjust: TitlecaseOptions = Default::default();
+/// no_adjust.head_adjustment = HeadAdjustment::NoAdjust;
+///
+/// // Exhibits head adjustment when set:
+/// assert_eq!(cm_normal.titlecase_segment_to_string("«hello»", &root, default_options), "«Hello»");
+/// assert_eq!(cm_legacy.titlecase_segment_to_string("«hello»", &root, default_options), "«Hello»");
+/// assert_eq!(cm_normal.titlecase_segment_to_string("«hello»", &root, no_adjust), "«hello»");
+///
+/// // Only changed in legacy mode:
+/// assert_eq!(cm_normal.titlecase_segment_to_string("丰(abc)", &root, default_options), "丰(abc)");
+/// assert_eq!(cm_legacy.titlecase_segment_to_string("丰(abc)", &root, default_options), "丰(Abc)");
+/// assert_eq!(cm_normal.titlecase_segment_to_string("丰(abc)", &root, no_adjust), "丰(abc)");
+///
+/// // Only changed in legacy mode:
+/// assert_eq!(cm_normal.titlecase_segment_to_string("49ers", &root, default_options), "49ers");
+/// assert_eq!(cm_legacy.titlecase_segment_to_string("49ers", &root, default_options), "49Ers");
+/// assert_eq!(cm_normal.titlecase_segment_to_string("49ers", &root, no_adjust), "49ers");
+/// ```
+/// <div class="stab unstable">
+/// 🚧 This code is experimental; it may change at any time, in breaking or non-breaking ways,
+/// including in SemVer minor releases. It can be enabled with the "experimental" Cargo feature
+/// of the icu meta-crate. Use with caution.
+/// <a href="https://github.com/unicode-org/icu4x/issues/2535">#2535</a>
+/// </div>
+#[derive(Clone, Debug)]
+pub struct TitlecaseMapper<CM> {
+    cm: CM,
+    gc: Option<DataPayload<GeneralCategoryV1Marker>>,
+}
+
+impl TitlecaseMapper<CaseMapper> {
+    /// A constructor which creates a [`TitlecaseMapper`], with the normal (non-legacy) head adjustment behavior.
+    /// See struct docs on [`TitlecaseMapper`] for more information on head adjustment behavior and usage examples.
+    ///
+    /// ✨ *Enabled with the `compiled_data` Cargo feature.*
+    ///
+    /// [📚 Help choosing a constructor](icu_provider::constructors)
+    #[cfg(feature = "compiled_data")]
+    pub const fn new() -> Self {
+        Self {
+            cm: CaseMapper::new(),
+            gc: Some(DataPayload::from_static_ref(
+                icu_properties::provider::Baked::SINGLETON_PROPS_GC_V1,
+            )),
```

Comment:
Thought: It looks like we decided to move forward with these cross-crate direct data dependencies until we have metrics that they are a problem.

Reply:
Yeah, also there's no easy way to depend on baked properties data and use the user APIs without needing to enum between CodePointSetData and CodePointSetDataBorrowed


(cc @robertbastian perhaps properties types should also be returning non-Borrowed payload types)

Reply:
This should be `icu_properties::maps::general_category().static_to_owned()`